src/helper/client.cpp

```diff
@@ -199,10 +204,55 @@ bool client::handle_command(network::request_init::request &command)
         return false;
     }
 
+    std::vector<std::shared_ptr<network::base_response>> messages;
+    messages.push_back(response);
+
     SPDLOG_DEBUG(
-        "sending response to request_init, verdict: {}", response.verdict);
+        "sending response to request_init, verdict: {}", response->verdict);
+    try {
+        return broker_->send(messages);
+    } catch (std::exception &e) {
+        SPDLOG_ERROR(e.what());
+    }
+
+    return false;
+}
+
+bool client::handle_command(network::config_sync::request & /* command */)
+{
+    if (!service_) {
+        // This implies a failed client_init, we can't continue.
+        SPDLOG_DEBUG("no service available on config_sync");
+        send_error_response(*broker_);
+        return false;
+    }
+
+    context_.emplace(*service_->get_engine());
```

Comment:
Why do you initialise a context here? This should only happen on `request_init` or `request_shutdown`

Reply:
Ok

Reply:
Done